File: touch/src/main/java/de/nicidienase/chaosflix/touch/browse/BrowseFragment.java
package de.nicidienase.chaosflix.touch.browse;

import android.arch.lifecycle.ViewModelProviders;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.View;

import de.nicidienase.chaosflix.R;
import de.nicidienase.chaosflix.touch.ViewModelFactory;

public class BrowseFragment extends Fragment {

	private BrowseViewModel mViewModel;

	@Override
	public void onCreate(@Nullable Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		mViewModel = ViewModelProviders.of(getActivity(), ViewModelFactory.INSTANCE).get(BrowseViewModel.class);

	}

	public BrowseViewModel getViewModel() {
		return mViewModel;
	}
}

File: touch/src/main/java/de/nicidienase/chaosflix/touch/browse/BrowseFragment.kt
package de.nicidienase.chaosflix.touch.browse

import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProvider
import androidx.lifecycle.ViewModelProviders
import de.nicidienase.chaosflix.common.viewmodel.BrowseViewModel
import de.nicidienase.chaosflix.common.viewmodel.ViewModelFactory

open class BrowseFragment : androidx.fragment.app.Fragment() {

    lateinit var viewModel: BrowseViewModel
    var overlay: View? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        viewModel = ViewModelProvider(requireActivity(), ViewModelFactory.getInstance(requireContext())).get(BrowseViewModel::class.java)
    }

    @JvmOverloads
    protected fun setupToolbar(toolbar: Toolbar, title: Int, isRoot: Boolean = true) {
        setupToolbar(toolbar, resources.getString(title), isRoot)
    }

    @JvmOverloads
    protected fun setupToolbar(toolbar: Toolbar, title: String, isRoot: Boolean = true) {
        val activity = activity as AppCompatActivity
        if (activity is BrowseActivity) {
            activity.setupDrawerToggle(toolbar)
        }
        activity.setSupportActionBar(toolbar)
        activity.supportActionBar?.title = title
        if (isRoot) {
            activity.supportActionBar?.setDisplayShowHomeEnabled(true)
        } else {
            activity.supportActionBar?.setDisplayHomeAsUpEnabled(true)
        }
    }

    protected fun setLoadingOverlayVisibility(visible: Boolean) {
        if (visible) {
            overlay?.visibility = View.VISIBLE
        } else {
            overlay?.visibility = View.INVISIBLE
        }
    }
}
